src/main/java/duke/CommandManager.java

```diff
@@ -0,0 +1,68 @@
+package duke;
+
+import duke.exceptions.CommandDescriptionEmptyException;
+import duke.exceptions.MissingFileException;
+import duke.exceptions.MissingParameterException;
+import duke.exceptions.UnknownCommandException;
+import duke.exceptions.*;
+import duke.tasks.Bye;
+import duke.tasks.Greet;
+import duke.tools.Parser;
+
+import java.io.IOException;
+
+/**
+ * Manage input commands.
+ * Keep reading input command until 'bye'.
+ */
+public class CommandManager {
+    private String inputCommand;
+    private String commandType;
+    private String commandDescription;
+    private static final TaskManager TASK_MANAGER = new TaskManager();
+    private static final Parser COMMAND_PARSER = new Parser();
+
+    public void setCommand(String command) throws UnknownCommandException, CommandDescriptionEmptyException {
+        this.inputCommand = command;
+
+        this.commandType = COMMAND_PARSER.getCommandType(command);
+        if(!(commandType.equals("todo")||commandType.equals("event")||commandType.equals("deadline")||commandType.equals("list")||commandType.equals("bye")||commandType.equals("mark")||commandType.equals("unmark")||commandType.equals("delete"))){
```

Comment:
This line looks unreadable, maybe you can create a separate method to check if the command is known.